PlayerUI/Views/PUIPlayerView.swift

```diff
@@ -353,20 +353,19 @@ public final class PUIPlayerView: NSView {
     }
     
     private func metadataBecameAvailable() {
+        guard let duration = self.asset?.duration else { return }
+
         DispatchQueue.main.async {
-            guard let duration = self.asset?.duration else { return }
-            
             self.timelineView.mediaDuration = Double(CMTimeGetSeconds(duration))
         }
     }
     
     fileprivate func playerTimeDidChange(time: CMTime) {
+        guard let player = player, player.hasValidMediaDuration,
```

Comment:
(Code style disagreement) I believe the original code was more clear and easier to change if needed.